src/api/affiliate.py

```diff
@@ -0,0 +1,35 @@
+"""Basic Flask API"""
+import os
+from typing import Any
+
+from dotenv import load_dotenv
+from dune_client.client import DuneClient
+from flask import Flask, jsonify
+
+from src.fetch.affiliate_data import CachingAffiliateFetcher
+from src.logger import set_log
+
+app = Flask(__name__)
+load_dotenv()
+cached_fetcher = CachingAffiliateFetcher(
```

Comment:
You should be able to use flask's `APScheduler` to trigger a cache refresh periodically (e.g. once every 5 minutes): https://github.com/viniciuschiele/flask-apscheduler/blob/master/examples/jobs.py

This would allow us to get rid of cache invalidation altogether (which should simplify the code). Basically the cache never becomes invalid it only gets updated periodically (and serves what ever data is current). We should add a prometheus metric on the last updated timestamp and alert if the update wasn't triggered for too long.

This design will also make sure there is only ever one thing writing the cache which makes us have to worry much less about concurrency.